arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/floorfilter/FloorFilterView.kt

```diff
@@ -0,0 +1,1071 @@
+        geoViewHolder = null
+        drawInGeoView = false
+        floorFilterManager.clearGeoView()
+    }
+
+    /**
+     * Binds this [FloorFilterView] to the provided [geoView], or unbinds it when passing in null.
+     *
+     * @throws IllegalStateException if this FloorFilterView is currently added to a [GeoView]
+     * @since 100.13.0
+     */
+    fun bindTo(geoView: GeoView?) {
+        if (drawInGeoView) {
+            throw IllegalStateException("FloorFilterView is currently added to a GeoView")
+        }
+        if (geoView == null) {
+            if (this.floorFilterManager.geoView != null) {
+                floorFilterManager.clearGeoView()
+            }
+        } else {
+            val map = getGeoModel(geoView) ?: throw IllegalStateException("There is no GeoModel in the GeoView")
+            setupGeoView(geoView, map)
+        }
+    }
+
+    /**
+     * Scrolls the levels list to show the selected level.
+     *
+     * @since 100.13.0
+     */
+    fun scrollToSelectedLevel() {
+        val selectedFloorPosition = levelAdapter.getSelectedLevelPosition()
+        if (selectedFloorPosition > -1) {
+            floorsRecyclerView?.scrollToPosition(selectedFloorPosition)
+        }
+    }
+
+    /**
+     * Opens the dialog to show the sites and facilities.
+     *
+     * @since 100.13.0
+     */
+    fun openSiteFacilityView() {
+        siteFacilityView.show()
+        siteFacilityButton?.isSelected = true
+    }
+
+    /**
+     * Closes the sites and facilities dialog.
+     *
+     * @since 100.13.0
+     */
+    fun closeSiteFacilityView() {
+        siteFacilityView.close()
+        siteFacilityButton?.isSelected = false
+    }
+
+    // Setup functions
+
+    /**
+     * Sets up the [FloorFilterView] to work with the provided [geoView].
+     *
+     * @since 100.13.0
+     */
+    private fun setupGeoView(geoView: GeoView, map: GeoModel) {
+        floorFilterManager.setupMap(geoView, map) {
+            updateSiteFacilityButtonEnabled()
+        }
+    }
+
+    /**
+     * Disables the site/facility button if there are no sites or facilities in the map. Hides the
+     * site/facility button if [hideSiteFacilityButton] is true.
+     *
+     * @since 100.13.0
+     */
+    private fun updateSiteFacilityButtonEnabled() {
+        siteFacilityButton?.isEnabled = floorFilterManager.sites.isNotEmpty() || floorFilterManager.facilities.isNotEmpty()
+        if (hideSiteFacilityButton) {
+            siteFacilityButton?.visibility = View.GONE
+        } else {
+            siteFacilityButton?.visibility = View.VISIBLE
+        }
+        updateSeparatorVisible()
+    }
+
+    private fun updateSeparatorVisible() {
+        siteFacilityButtonSeparator?.visibility = if (levelAdapter.itemCount > 0 && !hideSiteFacilityButton) View.VISIBLE else View.GONE
+    }
+
+    private fun addDataChangeListeners() {
+        floorFilterManager.setOnLevelChangeListener {
```

Comment:
would change to property call 

```suggestion
        floorFilterManager.onLevelChangeListener = {
            levelAdapter.updateData()
            onSelectionChangeListener?.invoke()
        }
```